src/App.tsx

```diff
@@ -10,22 +10,34 @@ import { MyTimer } from './components/Timer';
 import { usePrevious } from './hooks/usePrevious';
 import MyButton from './components/MyButton';
 import { useMyCallback } from './hooks/useMyCallback';
+import UserData from './mock/userdata';
+import { User } from './models/user';
+import MyList from './components/MyList/MyList';
 
 function App() {
   const [showTime, setShowTime] = useState(true);
   const previousShowTime = usePrevious(showTime);
   const btnContent = useMemo(() => <>show time</>, []);
+  const [userData, setUserData] = useState<User[]>([]);
 
   const changeShowTime = useMyCallback(() => {
     setShowTime((prevValue) => !prevValue);
   }, []);
 
+  useEffect(() => {
+    // Fake fetching
+    setTimeout(() => {
+      setUserData(UserData);
+    }, 800);
+  });
```

Comment:
If you don't specify a dependency array, this effect will run on every single render, so if a parent component renders, this component will also re-render and fetch your data again.

Remember: 

- Giving it an empty array acts like componentDidMount as in, it only runs once.
- Giving it no second argument acts as both `componentDidMount` and `componentDidUpdate`, as in it runs first on mount and then on every re-render.
- Giving it an array as second argument with any value inside, eg , [variable1] will only execute the code inside your `useEffect` hook ONCE on mount, as well as whenever that particular variable (variable1) changes.

```suggestion
  }, []);
```

Reference:

- https://react.dev/learn/synchronizing-with-effects#step-2-specify-the-effect-dependencies

Reply:
I was aware of it but missed this one 😓